test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,329 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiexportendpointslice
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/core"
+	corev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+	topologyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/topology/v1alpha1"
+	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/cluster"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestAPIExportEndpointSlice(t *testing.T) {
```

Comment:
It looks like we are missing a happy-case scenario. An SP creates a `slices` and it gets assigned `all/default` endpoints.

Reply:
I think that we discussed this point earlier. It is in the private test case where shards are guaranteed not to be modified by any other test.

Reply:
The destructive scenario is different. I think it is about `checking if adding a shard that matches partition requirements results in adding endpoints to the slice`.

I think we are missing a default case - `when I add a slice will it be filled with endpoints that correspond to the current number of shards?`

The default case should work on single and multi-shard environment and should not  add, remove or modify shards.

Reply:
The scenario is covered here: https://github.com/kcp-dev/kcp/pull/2608/files#diff-43025ba07d287623f79a84eaa3f3616ca31e4402bd6326683517094a172261efR289-R309
I don't see the point in making the shared cluster test fragile on purpose.

Reply:
oh, sorry I missed it. Wouldn't it be better to have scenarios that have well defined limited scope :) ?